lib/publish.js

```diff
@@ -1,43 +1,66 @@
 const execa = require('execa')
+const semver = require('semver')
 
 /**
  * @typedef {import('./index').Context} Context
  */
 /**
- * Push new docker image.
+ * Tag and push new docker images with semantic versioning.
  *
  * @param {Object} pluginConfig - The plugin configuration.
  * @param {Context} context - The semantic-release context.
  * @returns {Promise} A `Promise` that resolve to docker push command.
  */
 module.exports = async (pluginConfig, { nextRelease: { version }, logger }) => {
-  try {
+  /**
+   * Tag and push docker image by version.
+   *
+   * @param {string} version - Tag version.
+   * @returns {Promise} A `Promise` that resolve to docker push command.
+   */
+  async function publish (version) {
     logger.log(
       `Pushing version ${
         process.env.CI_REGISTRY_IMAGE
       }:${version} to GitLab Container Registry`
     )
-
-    // Push both new version and latest
+    // tag image
     await execa(
       'docker',
-      [
-        'tag',
-        `${process.env.CI_REGISTRY_IMAGE}:latest`,
-        `${process.env.CI_REGISTRY_IMAGE}:${version}`
-      ],
+      ['tag', `${process.env.CI_REGISTRY_IMAGE}:${version}`],
       {
         stdio: 'inherit'
       }
     )
+    // push image
     await execa(
       'docker',
       ['push', `${process.env.CI_REGISTRY_IMAGE}:${version}`],
       { stdio: 'inherit' }
     )
-    await execa('docker', ['push', `${process.env.CI_REGISTRY_IMAGE}:latest`], {
-      stdio: 'inherit'
-    })
+  }
+
+  try {
+    // parse version to parts
+    const { major, minor, patch, prerelease } = semver(version)
+    const isProdRelease = prerelease.length === 0
+
+    // first release version as it is
+    await publish(version)
+
+    if (isProdRelease) {
+      // on production release X:latest, X:1.1, X:1
+      await publish(`${major}.${minor}`)
+      await publish(`${major}`)
+      await publish('latest')
+      return
+    }
+
+    const [channel] = prerelease
+
+    // on other channels release X:channel, X:1.1.1-channel
+    await publish(channel)
+    await publish(`${major}.${minor}.${patch}-${channel}`)
```

Comment:
I think this should go in the `else` clause to avoid errors with versions without channels (`1.2.3`)
```suggestion
    } else {
      const [channel] = prerelease

      // on other channels release X:channel, X:1.1.1-channel
      await publish(channel)
      await publish(`${major}.${minor}.${patch}-${channel}`)
    }
```

Reply:
Merged, thanks!